crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -243,11 +236,44 @@ impl AlgorithmUpdaterV1 {
         }
     }
 
+    fn update_rewards(&mut self, fee_wei: u64) {
+        let block_da_reward = self.da_portion_of_fee(fee_wei);
+        self.total_da_rewards_excess = self
+            .total_da_rewards_excess
+            .saturating_add(block_da_reward.into());
+    }
+
+    fn update_projected_cost(&mut self, block_bytes: u64) {
+        let block_projected_da_cost =
+            (block_bytes as u128).saturating_mul(self.latest_da_cost_per_byte);
+        self.projected_total_da_cost = self
+            .projected_total_da_cost
+            .saturating_add(block_projected_da_cost);
+    }
+
+    // Take the `fee_wei` and return the portion of the fee that should be used for paying DA costs
+    fn da_portion_of_fee(&self, fee_wei: u64) -> u64 {
```

Comment:
Do we need to add the same function for execution price?

Reply:
The execution price uses the gas fullness to adjust. I don't know what part of the algorithm would be interested in the gathered fee from exec.

Reply:
It would be nice to know just total execution profit for statistic, plus later we can transfer part of the execution profit to the da profit.

Reply:
When we add any sort of metric that will consume the method, then we can add it.